frontend/src/v5/ui/themes/theme.ts

```diff
@@ -0,0 +1,372 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { createMuiTheme } from '@material-ui/core/styles';
+import { TypographyOptions } from '@material-ui/core/styles/createTypography';
+
+export const COLOR = {
+	WHITE: '#fff',
```

Comment:
Could we theme-name this too? 

I'm not a fan of naming a colour for a colour in themes. It is also odd that we name every colour base on theme usage here apart from white.

This seems to be used in contained button text colour, outlined button background colour. Can we have a name for this in the pallet ? if we end up wanting to change it from white to say, grey, or if dark mode means we want the outline button background to become black but keeping the font of contained button white then it becomes problematic and the `WHITE` being referenced around the codebase can also  become confusing.

tagging @george-hadfield  as it might need his input here.

Reply:
I'm not sure what the answer is here @carmenfan as something like "primary-main-text" won't work as some of the text in the buttons are only white on hover. Any suggestions welcome!

Reply:
I think we need to add button in there somewhere, like `primary-button-text` or something, or we may get confused with just standard text around website

Reply:
@carmenfan @sanmont3drepo 
It seems to me that we have not come up with any specific proposal regarding this name that would satisfy everyone.
I am attaching a screen and link below as a reference to how white is defined in the Material-UI for pallets colors and in general.
<img width="346" alt="Zrzut ekranu 2021-08-17 o 12 13 31" src="https://user-images.githubusercontent.com/24941866/129708201-a0be306a-b129-4c01-ab6b-1e010f7b9c1a.png">
https://material-ui.com/customization/default-theme/#explore

In general, under this assumption, each of the groups of colors that we have defined may have its own `contrastText`, but here you can also see that they did not shy away from defining white as `common` color. It also seems to me that the frequency of white or black color in any project often makes it unnecessary to name them differently.

So maybe using for now: `PRIMARY_MAIN_CONTRAST`, would be enough and so it will  not bind this color name, neither to the button nor to the text, because in the project white is also used as the background of the elements.

Let me know if it makes sense to you and do we want to name it like this.

Reply:
Im ok with that!

Reply:
sure

Reply:
@george-hadfield  if we can make sure we update that on the figma pallet to that would be great